src/sdl/index.ts

```diff
@@ -141,6 +137,25 @@ export class SDL {
     }
   }
 
+  constructor(
+    public readonly data: BetaSdl<V>,
+    public readonly version: NetworkVersion = "beta2",
+    public readonly network: NetworkId = "mainnet"
+  ) {
+    this.validate();
+  }
+
+  private validate() {
+    Object.keys(this.data.profiles.compute).forEach(name => {
+      const { resources } = this.data.profiles.compute[name];
+      if ("gpu" in resources) {
+        SDL.validateGPU(name, resources.gpu);
+      }
+
+      SDL.validateStorage(name, resources.storage);
+    });
+  }
```

Comment:
For instance this is how denom validation can be added here:
```suggestion
   this.validateDenom();
}
validateDenom() {
    const denoms = this.groups()
      .flatMap(g => g.resources)
      .map(resource => resource.price.denom);
    const invalidDenom = denoms.find(denom => denom !== "uakt" && denom !== USDC_IBC_DENOMS[this.network]);
    if (invalidDenom) throw new Error(`Invalid denom: "${invalidDenom}". Only uakt and ${usdcDenom} are supported.`);
  }
    });
```